### PR TITLE
feat(gql-tag-operations-preset): Allow `useFragment` to take arrays and nullable values as arguments

### PR DESCRIPTION
**Pull request by izumin5210**
## Description

Related #7808, #7809 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

I considered adding configuration values such as `fragmentMasking.allowList` and `fragmentMasking.allowNullable`. However, I decided to accept list and nullable by default instead of adding the settings for the following reasons:

- Being able to mask a list is always a necessary feature
- There is no disadvantage in accepting nullable if the argument is non-nullable and the return type is also non-nullable.

### CODE REVIEW

**Review by n1ru4l — APPROVED** [review]
Thank you for this great contribution 😍

**Review by n1ru4l — CHANGES_REQUESTED** [review]
@izumin5210 Can you please re-generate and commit the updated fixtures of the example apps. In addition, it would also be great if you could add some fixtures to the test that use the changes of this PR

**Comment by izumin5210** [comment]
Thank you for your code review 😻 
 
I have updated this PR ✅ 

- 1a6920f Update examples
- 6cc9b84 Add testcase

File: packages/presets/gql-tag-operations/tests/gql-tag-operations.spec.ts [hunk 1]
```diff
@@ -1,5 +1,8 @@
 import { executeCodegen } from '@graphql-codegen/cli';
+import { mergeOutputs } from '@graphql-codegen/plugin-helpers';
 import '@graphql-codegen/testing';
+import { validateTs } from '@graphql-codegen/testing';
+import { readFile } from 'fs/promises';
```
Comment (n1ru4l) [inline packages/presets/gql-tag-operations/tests/gql-tag-operations.spec.ts:5]: This is not available in Node.js 12 😭

We want to drop support soon, but for now we have to resolve this failing CI

Comment (izumin5210) [reply packages/presets/gql-tag-operations/tests/gql-tag-operations.spec.ts]: oh... 🙈 
I fixed it! 53ef642

**Review by n1ru4l — APPROVED** [review]
🚀

**Pull request merged**